### PR TITLE
Refactor: remove global package vars

### LINKED ISSUE #288: Remove using package global variable
In go, it is not recommended to use global variables, unless it is a constant.

Because in go these global vars can be mutated conmcurrently, which may introduce race  conditions

FYI https://golangdocs.com/golang-global-variables

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #288

**Special notes for your reviewer**:

This is a large patch remove all global package vars to make kmesh more golang authentic 

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: daemon/options/options.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 The Kmesh Authors.
```
Comment (LiZhenCheng9527) [inline daemon/options/options.go:2]: 2024

File: daemon/options/cni.go [hunk 2]
```diff
@@ -1,5 +1,4 @@
-/*
- * Copyright 2023 The Kmesh Authors.
+/* Copyright 2023 The Kmesh Authors.
```
Comment (LiZhenCheng9527) [inline daemon/options/cni.go:1]:
```suggestion
/* Copyright 2024 The Kmesh Authors.
```

File: pkg/bpf/bpf.go [hunk 3]
```diff
@@ -24,6 +21,8 @@ import (
 	"os/exec"
 	"strings"
 
+	"kmesh.net/kmesh/daemon/options"
```
Comment (LiZhenCheng9527) [inline pkg/bpf/bpf.go:24]: Is the import specification here wrong?

File: pkg/bpf/bpf_kmesh_l4.go [hunk 4]
```diff
@@ -15,14 +15,12 @@
...
  * limitations under the License.
-
- * Author: lec-bit
- * Create: 2023-11-02
  */
 
 package bpf
 
 import (
+	"kmesh.net/kmesh/daemon/options"
```
Comment (LiZhenCheng9527) [inline pkg/bpf/bpf_kmesh_l4.go:23]: ditto

File: pkg/controller/ads/ads_processor_test.go [hunk 5]
```diff
@@ -41,7 +41,7 @@ func TestHandleCdsResponse(t *testing.T) {
 	test.InitBpfMap(t)
 	t.Cleanup(test.CleanupBpfMap)
 	t.Run("new cluster, cluster type is eds", func(t *testing.T) {
-		svc := NewServiceEvent()
+		svc := newProcessor()
```
Comment (LiZhenCheng9527) [inline pkg/controller/ads/ads_processor_test.go:44]:
```suggestion
		processor := newProcessor()
```

File: pkg/controller/ads/ads_processor_test.go [hunk 6]
```diff
@@ -58,17 +58,23 @@ func TestHandleCdsResponse(t *testing.T) {
...
+=======
+		assert.Equal(t, []string{"ut-cluster"}, svc.rqt.ResourceNames)
+		assert.Equal(t, svc.LastNonce.edsNonce, svc.rqt.ResponseNonce)
+		assert.Equal(t, svc.Cache.ClusterCache.GetApiCluster(cluster.Name).ApiStatus, core_v2.ApiStatus_UPDATE)
+>>>>>>> c1e5e30 (ads controller refactor)
 	})
 
 	t.Run("new cluster, cluster type is not eds", func(t *testing.T) {
-		svc := NewServiceEvent()
+		svc := newProcessor()
```
Comment (LiZhenCheng9527) [inline pkg/controller/ads/ads_processor_test.go:77]: ditto

File: pkg/controller/workload/bpfcache/factory.go [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package bpfcache
```
Comment (LiZhenCheng9527) [inline pkg/controller/workload/bpfcache/factory.go:1]: copyright

File: pkg/bpf/bpf.go (hunk 3 above)
Comment (hzxuzhonghu) [reply pkg/bpf/bpf.go:24]: should  report error

**Pull request merged**